monitoring/monitoring/monitoring.py

```diff
@@ -227,18 +242,76 @@ async def polling_loop(app):
         await asyncio.sleep(60)
 
 
+async def monitor_disks(app):
+    log.info(f'monitoring disks')
+    compute_client: ComputeClient = app['compute_client']
+
+    DISK_SIZE_GB_SUMMARY.clear()
+    DISK_STATES.clear()
+
+    for zone in app['zones']:
+        async for disk in await compute_client.list(f'/zones/{zone}/disks', params={'filter': '(labels.batch = 1)'}):
+            log.info(f'disk {disk}')
+            namespace = disk['labels']['namespace']
+            size_gb = int(disk['sizeGb'])
+
+            creation_timestamp_msecs = parse_timestamp_msecs(disk.get('creationTimestamp'))
+            last_attach_timestamp_msecs = parse_timestamp_msecs(disk.get('lastAttachTimestamp'))
+            last_detach_timestamp_msecs = parse_timestamp_msecs(disk.get('lastDetachTimestamp'))
+
+            labels = {'zone': zone,
+                      'namespace': namespace}
+
+            DISK_SIZE_GB_SUMMARY.labels(**labels).observe(size_gb)
+
+            if creation_timestamp_msecs is None:
+                DISK_STATES.labels(state='creating', **labels).inc()
+            elif last_attach_timestamp_msecs is None:
+                DISK_STATES.labels(state='created', **labels).inc()
+            elif last_attach_timestamp_msecs is not None and last_detach_timestamp_msecs is None:
+                DISK_STATES.labels(state='attached', **labels).inc()
+            elif last_attach_timestamp_msecs is not None and last_detach_timestamp_msecs is not None:
+                DISK_STATES.labels(state='detached', **labels).inc()
```

Comment:
I'm wary to clear and then increment in an async loop. Maybe it's not a realistic issue but an unfortunately timed scrape could give a very incorrect picture. Could you instead count the occurences per unique set of labels and then `observe` outside of the async loop? Then we don't have to `clear`. `DISK_SIZE_GB_SUMMARY` is fine as it is.